Lib/ctypes/test/test_byteswap.py

```diff
@@ -311,5 +311,68 @@ class S(Structure):
         s2 = struct.pack(fmt, 0x12, 0x1234, 0x12345678, 3.14)
         self.assertEqual(bin(s1), bin(s2))
 
+    def test_union_fields(self):
+        if sys.byteorder == "little":
+            base = BigEndianUnion
+        else:
+            base = LittleEndianUnion
+
+        class T(base):
+            pass
+        _fields_ = [("a", c_ubyte),
+                    ("b", c_byte),
+                    ("c", c_short),
+                    ("d", c_ushort),
+                    ("e", c_int),
+                    ("f", c_uint),
+                    ("g", c_long),
+                    ("h", c_ulong),
+                    ("i", c_longlong),
+                    ("k", c_ulonglong),
+                    ("l", c_float),
+                    ("m", c_double),
+                    ("n", c_char),
+
+                    ("b1", c_byte, 3),
+                    ("b2", c_byte, 3),
+                    ("b3", c_byte, 2),
+                    ("a", c_int * 3 * 3 * 3)]
+        T._fields_ = _fields_
+
+        # these fields do not support different byte order:
+        for typ in c_wchar, c_void_p, POINTER(c_int):
+            _fields_.append(("x", typ))
```

Comment:
The field appended here will not be removed and so will be available in the next iteration, which I assume is not what you want. I recommend using the `+` operator instead to avoid modifying the original list.

I would do something like this.
```python
fields = [
    ("a", c_ubyte),
    ("b", c_byte),
    ("c", c_short),
    ("d", c_ushort),
    ("e", c_int),
    ("f", c_uint),
    ("g", c_long),
    ("h", c_ulong),
    ("i", c_longlong),
    ("k", c_ulonglong),
    ("l", c_float),
    ("m", c_double),
    ("n", c_char),
    ("b1", c_byte, 3),
    ("b2", c_byte, 3),
    ("b3", c_byte, 2),
    ("a", c_int * 3 * 3 * 3),
]

for typ in (c_wchar, c_void_p, POINTER(c_int)):
    with self.assertRaises(TypeError):
        class T(base):
            _fields_ = fields + [("x", typ)]
```

Reply:
Thanks for the review!

https://github.com/python/cpython/blob/8d28e0a61edeb073406c03a102b1145542f7da63/Lib/ctypes/test/test_byteswap.py#L173-L206

test_struct_field_1 is nearly the same as the code you raised this comment on; will fix for the same issue.